File: Core/Entities/Warehouses/Warehouse.cs
using System.ComponentModel.DataAnnotations;

namespace Core.Entities
{
    public class Warehouse : BaseEntity
    {
        public int CountryId { get; set; }
        public Country Country { get; set; }

        [Required]
        [MaxLength(100)]      
        public string City { get; set; }

        [Required]
        [MaxLength(100)]
        public string Street { get; set; }

        [Required]
        [MaxLength(255)]
        public string Name { get; set; }

    }
}

File: angular-happykids/src/app/shared/models/warehouse.ts
export interface Warehouse {
    id: number;
    name: string;
    street: string;
    city: string;
    countryId: number;
    country: string;
}

export interface WarehouseCreateEdit {
    id: number;
    name: string;
    street: string;
    city: string;
    countryId: number;
}
